### PR TITLE
Top Nav

### PR DESCRIPTION
**Pull request by marksy**
New topnav component
screen shot 2018-04-12 at 14 32 06

* optional logo/company area
* optional search-box area
* option service title
* optional navigation (children)
* TopNav.asAnchor for child anchors

### CODE REVIEW

File: components/top-nav/src/index.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  active: PropTypes.number,
+  bgColor: PropTypes.string,
+  color: PropTypes.string,
+  company: PropTypes.node,
+  serviceTitle: PropTypes.node,
+  search: PropTypes.node,
+  children: PropTypes.node,
+};
+
+export default TopNav;
```
Comment (penx) [inline components/top-nav/src/index.js]: this file is huge, could you make an `atoms` folder containing one glamorous component per file - search-wrapper.js, right-hand-side.js etc. ?

Would make the core functionality a lot easier to read.

Comment (marksy) [reply components/top-nav/src/index.js]: yeah sure

File: components/top-nav/src/index.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  active,
+  bgColor,
+  color,
+  company,
+  serviceTitle,
+  search,
+  children,
+  ...props
+}) => (
+  <React.Fragment>
```
Comment (penx) [inline components/top-nav/src/index.js]: If we're using Fragments, peerDependencies should be set to `"react": ">=16.2"`

File: components/top-nav/src/index.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+          }
+        </LogoSearchWrapper>
+        <RightHandSide>
+          {serviceTitle}
+          {children &&
+          <React.Fragment>
+            <Button mb={0}>menu</Button>
+            <Ul serviceTitle={serviceTitle}>
+              {children.length && children.map ? (
+                children.map((child, i) => (
```
Comment (penx) [inline components/top-nav/src/index.js]: this is where the context api and render props would come in useful, perhaps create a github issue to research this and add a `TODO: #XXX` to the code

File: components/radio/src/index.js.orig [hunk 4]
```diff
@@ -0,0 +1,112 @@
+// https://github.com/alphagov/govuk-frontend/blob/master/src/components/radios/_radios.scss
+// https://github.com/alphagov/govuk_elements/blob/master/assets/sass/elements/_forms.scss
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import glamorous from 'glamorous';
+import { YELLOW } from 'govuk-colours';
+import { withWhiteSpace } from '@govuk-react/hoc';
+<<<<<<< HEAD
```
Comment (penx) [inline components/radio/src/index.js.orig]: conflict

File: components/top-nav/package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    "@govuk-react/button": "^0.1.27",
+    "@govuk-react/constants": "^0.1.26",
+    "@govuk-react/icons": "^0.1.27",
+    "@govuk-react/search-box": "^0.1.27",
+    "govuk-colours": "^1.0.3"
+  },
+  "peerDependencies": {
+    "glamorous": ">=4",
+    "prop-types": ">=15",
+    "react": ">=16.2"
```
Comment (penx) [inline components/top-nav/package.json]: need to update peerDependency of react to 16.2 in packages/govuk-react too, probably worth doing it everywhere with a find replace

File: components/search-box/src/index.js.orig [hunk 6]
```diff
@@ -0,0 +1,88 @@
+// https://govuk-static.herokuapp.com/component-guide/search
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import glamorous from 'glamorous';
+<<<<<<< HEAD
```
Comment (penx) [inline components/search-box/src/index.js.orig]: conflict

Comment (marksy) [reply components/search-box/src/index.js.orig]: sorry

File: components/top-nav/package.json (hunk 5 above)
Comment (marksy) [reply components/top-nav/package.json]: done.

**Comment by penx** [comment]
I think it would be good to get this merged in as-is for now, but with any remaining work marked as issues or todo.

@Loque- could you please review and make any changes you think are needed before merge?

File: components/top-nav/src/atoms/menu-button/index.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+const Label = styled('label')({
+  [MEDIA_QUERIES.LARGESCREEN]: {
+    display: 'none',
+  },
+});
+
+const Input = styled('input')({
+  position: 'absolute',
+  top: '-999em',
+  // This is reliant on HTML structure of parent component.
```
Comment (Loque-) [inline components/top-nav/src/atoms/menu-button/index.js]: Edit: I would really like it if we could support React Native and we can look at improving things like this when we get the time to.

File: packages/constants/src/index.js [hunk 8]
```diff
@@ -33,6 +33,7 @@ export const SPACING = {
   SCALE_6: '60px',
 };
 
+// TODO: figure out how to optionally include locally installed font, e.g. "GDS Transport Website"
```
Comment (Loque-) [inline packages/constants/src/index.js]: Could it not be part of the font stack?

Edit; ah oki it already is, but does not work if you have the font locally installed?

Comment (penx) [reply packages/constants/src/index.js]: only if you have it installed as 'nta' - the one I have is called something else locally

**Pull request merged**